ext/simplexml/simplexml.c

```diff
@@ -2615,7 +2615,7 @@ PHP_FUNCTION(simplexml_import_dom)
 	zend_class_entry *ce = sxe_class_entry;
 	zend_function    *fptr_count;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "o|C!", &node, &ce) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|C!", &node, sxe_class_entry, &ce) == FAILURE) {
 		RETURN_THROWS();
 	}
 
```

Comment:
Or maybe the param type in the stub should be adapted to `object` and the `object = Z_LIBXML_NODE_P(node);` call should be moved inside `if (nodep && nodep->type == XML_ELEMENT_NODE) {`?

Reply:
Based on the tests (e.g. ext/simplexml/tests/bug51615.phpt) it looks like this PR in its current form is not appropriate